server/app/controllers/applicant/ProgramSlugHandler.java

```diff
@@ -149,6 +151,11 @@ private Result redirectToOverviewOrReviewPage(
       long applicantId,
       ProgramDefinition activeProgramDefinition,
       ApplicationPrograms relevantPrograms) {
+    // External programs don't have an overview or review page
+    if (activeProgramDefinition.programType().equals(ProgramType.EXTERNAL)) {
+      return Results.badRequest(new ProgramNotFoundException(programSlug).getMessage());
```

Comment:
Used `ProgramNotFoundException` error to mimic the behavior when an invalid program url is provided. I think it would be less confusing for the user to see "this program doesn't exist" than a longer explanation of "this is an external program and it's information is outside civiform" (reworded). 
Specially because the programId is not exposed to the applicants, so for them to get to this page would be unlikely.
Now, the administrators could go to that link thinking it will show something.. but I think having a better message  on the admin panel explaining program URL is not exposed for external sites would be better.

Thoughts?

Reply:
I think given this URL isn't exposed anywhere and there aren't any links in the admin panel it's probably fine that we don't show a helpful error message. Maybe its worth creating another exception called ProgramLinkUnsupportedException or something which is a more accurate exception though?

Reply:
`ProgramNotFoundException` along with some other exceptions gets treated special in the [ErrorHandler class](https://github.com/civiform/civiform/blob/b2d86f32d7523f30543b4438a3215e0aabd8164c/server/app/controllers/ErrorHandler.java#L103) which makes it not show the NotFound page.

If we create a new exception it should be added to that list to behave the same way.

I suggest we just keep the `ProgramNotFoundException` rather than adding a new exception for this one thing.

We should eventually revisit how the system as a whole handles that list of exceptions and maybe that should get an issue created.

Reply:
Agreed we should probably revisit exceptions as a whole. But yeah I guess this exception name isn't wrong. Going to approve

Reply:
Yeah, I was also debating whether to create a new error but seemed okay to use `ProgramNotFoundException` since for the user trying to navigate to an invalid programId is effectively "program not found".